### PR TITLE
fix(helm): add annotations key to allow for annotations

### PR DESCRIPTION
**Pull request by sicoyle**
# Description

backport of https://github.com/dapr/dapr/pull/7920

## Issue reference

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by cicoyle — DISMISSED** [review]

File: charts/dapr/charts/dapr_operator/templates/dapr_operator_service.yaml [hunk 1]
```diff
@@ -7,9 +7,10 @@ metadata:
     {{- range $key, $value := .Values.global.k8sLabels }}
     {{ $key }}: {{ tpl $value $ }}
     {{- end }}
-{{- if .Values.apiService.annotations }}
-{{ toYaml .Values.apiService.annotations | indent 4}}
-{{- end }}
+  {{- if .Values.apiService.annotations }}
+  annotations:
+  {{ toYaml .Values.apiService.annotations | indent 4}}
```
Comment (ItalyPaleAle) [inline charts/dapr/charts/dapr_operator/templates/dapr_operator_service.yaml:12]: You must add `-` at the start or the first line will have indentation of 6 (and the second of 4)
```suggestion
  {{- toYaml .Values.apiService.annotations | indent 4}}
```

File: charts/dapr/charts/dapr_placement/templates/dapr_placement_service.yaml [hunk 2]
```diff
@@ -9,9 +9,10 @@ metadata:
     {{- range $key, $value := .Values.global.k8sLabels }}
     {{ $key }}: {{ tpl $value $ }}
     {{- end }}
-{{- if .Values.service.annotations }}
-{{ toYaml .Values.service.annotations | indent 4}}
-{{- end }}
+  {{- if .Values.service.annotations }}
+  annotations:
+  {{ toYaml .Values.service.annotations | indent 4}}
```
Comment (ItalyPaleAle) [inline charts/dapr/charts/dapr_placement/templates/dapr_placement_service.yaml:14]: Same

```suggestion
  {{- toYaml .Values.service.annotations | indent 4}}
```

File: charts/dapr/charts/dapr_sentry/templates/dapr_sentry_service.yaml [hunk 3]
```diff
@@ -7,9 +7,10 @@ metadata:
     {{- range $key, $value := .Values.global.k8sLabels }}
     {{ $key }}: {{ tpl $value $ }}
     {{- end }}
-{{- if .Values.service.annotations }}
-{{ toYaml .Values.service.annotations | indent 4}}
-{{- end }}
+  {{- if .Values.service.annotations }}
+  annotations:
+  {{ toYaml .Values.service.annotations | indent 4}}
```
Comment (ItalyPaleAle) [inline charts/dapr/charts/dapr_sentry/templates/dapr_sentry_service.yaml:12]:
```suggestion
  {{- toYaml .Values.service.annotations | indent 4}}
```

File: charts/dapr/charts/dapr_sidecar_injector/templates/dapr_sidecar_injector_service.yaml [hunk 4]
```diff
@@ -8,9 +8,10 @@ metadata:
     {{- range $key, $value := .Values.global.k8sLabels }}
     {{ $key }}: {{ tpl $value $ }}
     {{- end }}
-{{- if .Values.service.annotations }}
-{{ toYaml .Values.service.annotations | indent 4}}
-{{- end }}
+  {{- if .Values.service.annotations }}
+  annotations:
+  {{ toYaml .Values.service.annotations | indent 4}}
```
Comment (ItalyPaleAle) [inline charts/dapr/charts/dapr_sidecar_injector/templates/dapr_sidecar_injector_service.yaml:13]:
```suggestion
  {{- toYaml .Values.service.annotations | indent 4}}
```

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]

**Comment by cicoyle** [comment]
I've adding @ItalyPaleAle's suggestions. lgtm

**Review by JoshVanL — APPROVED** [review]

**Pull request closed without merging**